src/Chapter3.hs

```diff
@@ -971,6 +1064,18 @@ implement the following functions:
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
 
+data DayOfWeek = Monday | Tuesday | Wednesday | Thursday | Friday | Saturday | Sunday deriving (Show, Enum, Eq, Ord)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend d = d == Saturday || d == Sunday
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay Sunday = Monday
+nextDay d = succ d
+
+daysToParty :: DayOfWeek -> Int
+daysToParty d = mod (fromEnum Friday - fromEnum d + 7) 7
```

Comment:
Nice math solution! :1234: Though, you don't even need to add 7. `mod` already works properly for negative numbers:

```haskell
ghci> mod (-1) 7
6
```

Reply:
Good to know. I used to `mod` producing negative numbers in C# and C++ and just implement that same pattern elsewhere

Reply:
Fixed in 7c01273